wavefront/client.go

```diff
@@ -0,0 +1,209 @@
+package wavefront
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/wavefronthq/wavefront-sdk-go/event"
+	"github.com/wavefronthq/wavefront-sdk-go/histogram"
+	"github.com/wavefronthq/wavefront-sdk-go/internal"
+	"github.com/wavefronthq/wavefront-sdk-go/types"
+)
+
+// Client Interface for sending metrics, distributions and spans to Wavefront
+type Client interface {
```

Comment:
Let's not introduce a new `Client` interface. `WavefrontSender` will still be the main interface.

`Client` will be the singular implementation (struct) of `WavefrontSender` (instead of the previous direct vs proxy clients). And MultiClient likewise a wrapper around clients that implements the WavefrontSender interface.